src/BuiltInTools/dotnet-watch/HotReloadDotNetWatcher.cs

```diff
@@ -378,20 +385,44 @@ await Task.WhenAll(
 
                         async Task<ImmutableList<ChangedFile>> CaptureChangedFilesSnapshot(ImmutableDictionary<ProjectId, string>? rebuiltProjects)
                         {
-                            var changedFiles = Interlocked.Exchange(ref changedFilesAccumulator, []);
-                            if (changedFiles is [])
+                            var changedPaths = Interlocked.Exchange(ref changedFilesAccumulator, []);
+                            if (changedPaths is [])
                             {
                                 return [];
                             }
 
+                            // Note:
+                            // It is possible that we could have received multiple changes for a file that should cancel each other (such as Delete + Add),
```

Comment:
I think Add then Delete is generally "nothing happened here".  I think Delete then Add is equivalent to "Update".  I'm thinking in particular about how something like the Resx code generator might work to product <ResxFileName>.Designer.cs from <ResxFileName>.resx, which could be Delete then Add.  Does your change reflect DeleteThenAdd => Update?  Or is that a not-hittable scenario?

Reply:
Yup, delete+add translates to update. See https://github.com/dotnet/sdk/pull/45131/files#diff-703e093881ff7958e3d7a302b86e8a2c4c7b61825e6e8a379d743f93b6def1f3R17 for all cases.